lib/charms/layer/azure.py

```diff
@@ -306,87 +333,89 @@ def create_loadbalancer(request):
             "--resource-group",
             resource_group,
+            "--tags",
+            model_tag,
         )
 
-    if request.get("backend_address"):
-        for i, backend_address in request.get("backend_address"):
-            lb_create_args += [
-                "--backend-pool-name",
-                "{}-ip-{}".format(request.get("name"), i),
-            ]
-            # nic = _get_nic_from_ip(backend_address, resource_group)
-            _azure(
-                "network",
-                "lb",
-                "address-pool",
-                "create",
-                "--name",
-                "{}-ip-{}".format(request.get("name"), i),
-                "--resource-group",
-                resource_group,
-                "--lb-name",
-                request.get("name"),
-                "--backend-address",
-                backend_address,
-            )
-
-    lb_created = _azure("network", *lb_create_args)
+    backend_args = []
+    for i, backend in enumerate(request.backends):
+        backend_args += [
+            "--backend-address",
+            "name=addr{}".format(i),
+            "ip-address={}".format(backend),
+        ]
+    _azure(
+        "network",
+        "lb",
+        "address-pool",
+        "create",
+        "--name",
+        lb_pool_name,
+        "--resource-group",
+        resource_group,
+        "--lb-name",
+        lb_name,
+        *backend_args,
+    )
 
-    for front, back in request.get("port_mapping").items():
+    for front, back in request.port_mapping.items():
         _azure(
             "network",
             "lb",
             "rule",
             "create",
             "--name",
-            "{}-rule-{}-{}".format(request.get("name"), front, back),
+            "{}-rule-{}-{}".format(lb_name, front, back),
             "--resource-group",
             resource_group,
             "--lb-name",
-            request.get("name"),
+            lb_name,
             "--frontend-port",
-            "{}".format(front),
+            str(front),
             "--backend-port",
-            "{}".format(back),
+            str(back),
             "--protocol",
-            request.get("protocol"),
+            request.protocol.value.capitalize(),
         )
 
-    for i, health_check in enumerate(request.get("health_checks")):
+    for i, health_check in enumerate(request.health_checks):
         lb_probe_create_args = [
             "lb",
             "probe",
             "create",
             "--name",
-            "{}-probe-{}".format(request.get("name"), i),
+            "{}-probe-{}".format(lb_name, i),
             "--resource-group",
             resource_group,
             "--lb-name",
-            request.get("name"),
+            lb_name,
             "--protocol",
-            health_check.get("type"),
+            health_check.protocol.value.capitalize(),
             "--port",
-            health_check.get("port"),
+            health_check.port,
```

Comment:
I see in the schema that [port is an int field](https://github.com/juju-solutions/loadbalancer-interface/blob/5e477eee6702cae94c28574f02c4f39b27068e31/loadbalancer_interface/schemas/v1.py#L80). Does `health_check.port` match that type? If so, I think you need to wrap this in `str()`, otherwise it will get passed through [_azure](https://github.com/charmed-kubernetes/charm-azure-integrator/blob/4619e62ac487682d0c71185a261c9dfc70f821ff/lib/charms/layer/azure.py#L572) to subprocess.run and cause a TypeError.